Library/Homebrew/os/mac/mach.rb

```diff
@@ -155,6 +155,13 @@ def ppc64?
     arch == :ppc64
   end
 
+  def arch_compatible?(wanted_arch)
+    return true unless mach_data.present?
+    return arch == wanted_arch unless universal?
+
+    archs.include?(wanted_arch)
+  end
```

Comment:
I wonder if we should just make this always true on macOS given it's never been problematic? Linux has separate `ld` loaders but macOS does not.

Reply:
Done in 6329db9065ae670f0f69454b3e58ec81ba7b3ea3